elastic/kibana
# [AO] Add alertDetailsUrl to infra rule types

Closes #156534

## Summary

This PR adds the alertDetailsUrl to the infra rules. The value of this variable is a link to the `observability > alerts` page filtered for this instance of alert.

![image](https://github.com/elastic/kibana/assets/12370520/409bea90-5d2b-4e60-ae4c-61223cccd41a)

Here is an example of this action variable:

|alertDetailsUrl as action variable|Result of action|
|---|---|
|![image](https://github.com/elastic/kibana/assets/12370520/4d920710-357a-4557-b871-36f0f8ecf0be)|![image](https://github.com/elastic/kibana/assets/12370520/4f800c6d-f15f-481e-b7fc-4f85aa1085a7)|

**Note**
- I will change this field to `kibana.alert.url` in another [ticket](https://github.com/elastic/kibana/issues/158359)

## 🧪 How to test
- Ensure that `server.publicBaseUrl` is configured in kibana.dev.yml
- Create a metric threshold/inventory/logs rule and use the `context.alertDetailsUrl` in action for this rule
- After an alert is triggered, open the link provided by alertDetailsUrl and make sure that the alert is filtered correctly
- Check the time range, it should be set for 5 mins before the alert start time
